plugins/git/neogit/default.nix

```diff
@@ -102,6 +102,13 @@ helpers.neovim-plugin.mkNeovimPlugin config {
     };
   };
 
+  extraOptions = {
+    whichPackage = helpers.mkPackageOption {
```

Comment:
> I wonder if for neogit we need a more freeform approach, e.g. `attrsOf (nullOr package)`.
> 
> Or at least a `whichPackage` option.
> 
> Saying that, is it even possible to do anything without the `which` command? Installing the dependency seems a little overkill. Maybe it's for cross-platform reasons 🤷

@MattSturgeon  I'm not familiar with why the dependency was added... i might have to look into it

Reply:
I'm also unfamiliar, it was not present [before migrating](https://github.com/nix-community/nixvim/blob/82a24d0d422bcc24eb33ebe7aa2571a78f4eafdf/plugins/git/neogit.nix) to mkNeovimPlugin, so it was added in #971

AFACT, it does no harm. And could lead to better consistency on platforms that use a shell builtin or a different implementation of `which`?

cc @GaetanLepage as original implementor